child_compassion/models/child_compassion.py

```diff
@@ -552,18 +554,22 @@ def fetch_missing_relational_records(self, field_relation, values, json_name):
         # to take from onramp result
         for i, value in enumerate(values):
             # check if hobby/household duty, etc... exists in our database
-            search_vals = [("name", "=", value)]
+            search_vals = [(field_name, "=", value)]
             relation_obj = self.env[field_relation].sudo()
             if hasattr(relation_obj, "value"):
+                # Useful for connect.multipicklist objects
                 search_vals.insert(0, "|")
                 search_vals.append(("value", "=", value))
             search_count = relation_obj.search_count(search_vals)
             # if not exist, then create it
             if not search_count:
                 value_record = (
-                    relation_obj.create({"name": value, "value": value})
+                    relation_obj.create({field_name: value, "value": value})
```

Comment:
Need to check again if relation_obj has attribute "value" if we want to create a new record with "value": value. Otherwise, create record with column "name" only.

Reply:
This is not needed here because the record creation will work and there will only be a warning in the console telling that the field "value" was ignored. It's not like in the search where it raises an error in the search method.